src/matrixrtc/CallMembership.ts

```diff
@@ -149,99 +138,50 @@ export class CallMembership {
-
-        if ("expires" in this.membershipData) {
-            // we know createdTs exists since we already do the isLegacyCallMembershipData check
-            return this.createdTs() + this.membershipData.expires;
-        } else {
-            // We know it exists because we checked for this in the constructor.
-            return this.membershipData.expires_ts;
-        }
-    }
-
-    /**
-     * Gets the expiry time of the event, converted into the device's local time.
-     * @deprecated This function has been observed returning bad data and is no longer used by MatrixRTC.
-     * @returns The local expiry time of the membership as a unix timestamp in milliseconds or undefined if not applicable
-     */
-    public getLocalExpiry(): number | undefined {
-        // if the membership is not a legacy membership, we assume it is MSC4143
-        if (!isLegacyCallMembershipData(this.membershipData)) return undefined;
-
-        if ("expires" in this.membershipData) {
-            // we know createdTs exists since we already do the isLegacyCallMembershipData check
-            const relativeCreationTime = this.parentEvent.getTs() - this.createdTs();
-
-            const localCreationTs = this.parentEvent.localTimestamp - relativeCreationTime;
-
-            return localCreationTs + this.membershipData.expires;
-        } else {
-            // With expires_ts we cannot convert to local time.
-            // TODO: Check the server timestamp and compute a diff to local time.
-            return this.membershipData.expires_ts;
-        }
+    public getAbsoluteExpiry(): number {
+        // TODO: calculate this from the MatrixRTCSession join configuration directly
+        return this.createdTs() + (this.membershipData.expires ?? DEFAULT_EXPIRE_DURATION);
     }
 
     /**
      * @returns The number of milliseconds until the membership expires or undefined if applicable
      */
-    public getMsUntilExpiry(): number | undefined {
-        if (isLegacyCallMembershipData(this.membershipData)) {
-            // Assume that local clock is sufficiently in sync with other clocks in the distributed system.
-            // We used to try and adjust for the local clock being skewed, but there are cases where this is not accurate.
-            // The current implementation allows for the local clock to be -infinity to +MatrixRTCSession.MEMBERSHIP_EXPIRY_TIME/2
-            return this.getAbsoluteExpiry()! - Date.now();
-        }
-
-        // Assumed to be MSC4143
-        return undefined;
+    public getMsUntilExpiry(): number {
+        // Assume that local clock is sufficiently in sync with other clocks in the distributed system.
+        // We used to try and adjust for the local clock being skewed, but there are cases where this is not accurate.
+        // The current implementation allows for the local clock to be -infinity to +MatrixRTCSession.MEMBERSHIP_EXPIRY_TIME/2
+        return this.getAbsoluteExpiry() - Date.now();
```

Comment:
It looks like the impl of this is taking the legacy bit from above? I would expect it to be the other way around, although that bit just returns 'undefined'...

Reply:
Two things have got conflated here:

1. Removing the legacy logic
2. Implementing an expiry logic for the non-legacy code

I've removed the latter from this PR to make the review easier. It should go in another PR. What I have done is left some TODOs and commented out code in to help with this. If you would prefer that they are removed outright then we can do so.

Reply:
Gotcha, thanks. I think it was simple enough this would have been okay, it just wasn't obvious what was going on from reading through. Thanks though.